spec-tests/src/test/java/org/candlepin/spec/hypervisors/HypervisorCheckInSpecTest.java

```diff
@@ -1613,10 +1613,15 @@ private HypervisorUpdateResultDTO getResultData(AsyncJobStatusDTO status) throws
 
     private void confirmResultDataCounts(HypervisorUpdateResultDTO resultData, int created, int updated,
         int unchanged, int failedUpdate) {
-        assertEquals(created, resultData.getCreated().size());
-        assertEquals(updated, resultData.getUpdated().size());
-        assertEquals(unchanged, resultData.getUnchanged().size());
-        assertEquals(failedUpdate, resultData.getFailedUpdate().size());
+        int createdCount = resultData.getCreated() == null ? 0 : resultData.getCreated().size();
```

Comment:
Or more to the point: what has the hypervisor result data behaviour been up until now? were these Sets always returned as empty arrays, or as null by default? Because this seems like the API changed, and we're changing the test to match, which does not look right

Reply:
Looking back at old code, the server has always been returning these as empty arrays: https://github.com/candlepin/candlepin/blob/candlepin-2.3-HOTFIX/server/src/main/java/org/candlepin/resource/dto/HypervisorUpdateResult.java#L42-L46 so we need to make sure that continues, and this test should be reverted to always expect them as non-null, empty Sets